src/praisonai-train/tests/unit/data/test_benchmark.py

```
@@ -0,0 +1,215 @@
+
+def test_failed_response_does_not_add_tokens(monkeypatch):
+    # A completion that reports tokens but has empty content is ok=False AND
+    # contributes 0 tokens, so token throughput stays consistent with success.
+    payload = {"usage": {"completion_tokens": 99},
+               "choices": [{"message": {"content": ""}}]}
+
+    def _fake_urlopen(req, timeout=None):
+        return io.BytesIO(json.dumps(payload).encode())
+
+    monkeypatch.setattr(bm.urllib.request, "urlopen", _fake_urlopen)
+    dt, ctok, ok = bm._timed_call({**_cfg(), "deployment": "d"}, [], 5)
```

Ruff (0.15.21)

[warning] Unpacked variable `dt` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)